### PR TITLE
Update free-programming-books-pt_BR.md

### PR DESCRIPTION
**Pull request by rickytaki**
Caelum and K19 are popular searches, so I updated the authors reference.

Changed links that require login to a separate category named Account required, as issue #1223

### CODE REVIEW

File: free-programming-books-pt_BR.md [hunk 1]
```diff
@@ -19,42 +17,35 @@
...
 * [Shell Script](#shell)
 * [Vim](#vim)
+* [Account Required (Necessário ter conta)](#account-required)
+  * [Metodologias de Desenvolvimento de Software](#metodologias-de-desenvolvimento-de-software)
 
 
 ### Android
 
-* [Desenvolvimento Mobile com Android](http://www.k19.com.br/downloads/apostilas/java/k19-k41-desenvolvimento-mobile-com-android)
+* [Desenvolvimento Mobile com Android - K19](http://www.k19.com.br/downloads/apostilas/java/k19-k41-desenvolvimento-mobile-com-android)
```
Comment (vhf) [inline free-programming-books-pt_BR.md]: K19 should go after the link to be consistent

**Comment by vhf** [comment]
Thanks for your work!

> Changed links that require login to a separate category named Account required, as issue #1223

Hmm, I probably was not clear enough in my last message on this issue. I would prefer if all these links stayed in place, but with `*(account required)*` at the end of the line, to warn people.

**Comment by rickytaki** [comment]
Ok, np!! I'll fix ASAP

**Comment by rickytaki** [comment]
Could you check if I nailed it this time?? :)
If there is anything else I can help, let me know! I can guarantee the pt-BR file, but if you need more help,  can look the EN file too...

File: free-programming-books-pt_BR.md [hunk 2]
```diff
@@ -23,38 +23,38 @@
...
 
 
 ### C
 
-* [Apostila Linguagem C - Silvio Lago](http://www.ime.usp.br/~slago/slago-C.pdf) (PDF)
+* [Apostila Linguagem C](http://www.ime.usp.br/~slago/slago-C.pdf) - Silvio Lago (PDF)
 * [Introdução a Programação](https://github.com/ufpb-computacao/introducao-a-programacao-livro/releases) - livro adotado na UFPB.
 * [Introdução ao Curso de C ](http://www.ic.unicamp.br/~mc102/introducao-ao-curso-de-c.html).
-* [O Fantástico Mundo da Linguagem C](https://fiorix.wordpress.com/2014/04/12/livro-sobre-c-e-linux/)
+* [O Fantástico Mundo da Linguagem C](https://fiorix.files.wordpress.com/2014/04/o-fantc3a1stico-mundo-da-linguagem-c.pdf)
```
Comment (vhf) [inline free-programming-books-pt_BR.md]: Please add (PDF) at the end of this line

File: free-programming-books-pt_BR.md [hunk 3]
```diff
@@ -74,26 +74,26 @@
...
-* [Desenvolvimento Web com JSF 2 e JPA 2 - K19](http://www.k19.com.br/downloads/apostilas/java/k19-k12-desenvolvimento-web-com-jsf2-e-jpa2)
-* [Desenvolvimento Web com Struts - K19](http://www.k19.com.br/downloads/apostilas/java/k19-k52-desenvolvimento-web-com-struts)
-* [Design Patterns em Java - K19](http://www.k19.com.br/downloads/apostilas/java/k19-k51-design-patterns-em-java)
-* [Integração de Sistemas com Webservices, JMS e EJB - K19](http://www.k19.com.br/downloads/apostilas/java/k19-k23-integracao-de-sistemas-com-webservices-jms-e-ejb)
+* [Algoritmos e Estruturas de Dados com Java](http://www.caelum.com.br/apostila-java-estrutura-dados/) - Caelum
+* [Desenvolvimento Web Avançado com JSF 2, EJB 3.1 e CDI](http://www.k19.com.br/downloads/apostilas/java/k19-k22-desenvolvimento-web-avancado-com-jsf2-ejb3.1-e-cdi) - k19
+* [Desenvolvimento Web com JSF 2 e JPA 2](http://www.k19.com.br/downloads/apostilas/java/k19-k12-desenvolvimento-web-com-jsf2-e-jpa2) - K19
+* [Desenvolvimento Web com Struts](http://www.k19.com.br/downloads/apostilas/java/k19-k52-desenvolvimento-web-com-struts)  - K19
```
Comment (vhf) [inline free-programming-books-pt_BR.md]: Please leave only one space between `)` and `- K19`

**Comment by vhf** [comment]
This is great, thanks a lot for your work.

I added two nitpicks to your latest commit, and I'll merge as soon as this is fixed.

**Comment by rickytaki** [comment]
Done!

**Pull request merged**